### PR TITLE
remove unused/untested methods

### PR DESCRIPTION
**Pull request by grosser**
auditing_enabled was especially dangerous since it sets on the instance and modifies the global state

@tbrisker broken out from https://github.com/collectiveidea/audited/pull/356

### CODE REVIEW

**Comment by grosser** [comment]
@tbrisker I can haz review plz

File: lib/audited/auditor.rb [hunk 1]
```diff
@@ -204,7 +200,7 @@ def audited_changes
         if audited_options[:only].present?
           all_changes.slice(*audited_columns)
         else
-          all_changes.except(*non_audited_columns)
+          all_changes.except(*self.class.non_audited_columns)
```
Comment (tbrisker) [inline lib/audited/auditor.rb]: Why is this change needed? why does it matter if we have an instance method that calls the class method just for convenience?

File: lib/audited/auditor.rb [hunk 2]
```diff
@@ -297,10 +293,6 @@ def run_conditional_check(condition, matching: true)
         true
       end
 
-      def auditing_enabled=(val)
-        self.class.auditing_enabled = val
-      end
```
Comment (tbrisker) [inline lib/audited/auditor.rb]: I'm OK with removing this method, but please add a line in the CHANGELOG file indicating this change and that the class method should be used instead.

File: lib/audited/auditor.rb (hunk 1 above)
Comment (grosser) [reply lib/audited/auditor.rb]: - having an instance method for a class variable makes it look like it's changeable/configured on the instance, so extra mental overhead and some users might think it's ok to do `instance.non_audited_columns << :foo`
 - it's not much convenience if it's only needed once
 - for symmetry also removed `def audited_columns `

File: lib/audited/auditor.rb (hunk 2 above)
Comment (grosser) [reply lib/audited/auditor.rb]: done
